packages/mermaid/src/diagrams/common/common.ts

```diff
@@ -208,21 +208,33 @@ export const parseGenericTypes = function (input: string): string {
   return output.join('');
 };
 
+export const countOccurrence = (string: string, substring: string): number => {
+  return Math.max(0, string.split(substring).length - 1);
```

Comment:
Can `string.split().length` ever be `0`? My feeling is that it can never be less than `1`, so the following should be safe.

```suggestion
  return string.split(substring).length - 1;
```

Reply:
It can, when the input string is empty. I added the max call while writing the unit test when the "" case was added.

Reply:
> It can, when the input string is empty. I added the max call while writing the unit test when the "" case was added.

:facepalm: That's my bad for reviewing your PR too quickly! You're right. `"".split("").length` returns `0`.

There's even a big warning about it on [`String.prototype.split()` MDN page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split):
![image](https://github.com/mermaid-js/mermaid/assets/19716675/36042193-3aa6-41bf-827a-17f81ee33ef3)

Great call on testing this edge case :smile: